staging/src/k8s.io/client-go/resolver/dns/resolver.go

```diff
@@ -0,0 +1,331 @@
+package dns
+
+import (
+	"context"
+	"net"
+
+	"golang.org/x/net/dns/dnsmessage"
+)
+
+const ttl = 300
+
+// MemResolver implement an in memory resolver that receives DNS questions and
+// executes the corresponding Lookup functions. If the corresponding Lookup
+// function is not present, it uses the DefaultResolver ones.
+type MemResolver struct {
+	LookupAddr  func(ctx context.Context, addr string) (names []string, err error)
+	LookupCNAME func(ctx context.Context, host string) (cname string, err error)
+	LookupHost  func(ctx context.Context, host string) (addrs []string, err error)
+	LookupIP    func(ctx context.Context, network, host string) ([]net.IP, error)
```

Comment:
can we shrink this to only have `LookupIP`? And make it local to the parent package. This code here is a lot.

Reply:
yeah, this was the code I've tried to get into golang :) I'll keep it to a minimum